src/components/articles/viewArticles.jsx

```diff
@@ -1,13 +1,119 @@
 import React, { Component } from "react";
+import { connect } from "react-redux";
+import { PropTypes } from "prop-types";
+import NavBar from "../common/navBar.jsx";
+import Pagination from "../common/pagination.jsx";
+import ArticleCard from "./articleCard.jsx";
+import { getArticles } from "../../actions/viewArticles";
+import { paginate } from '../../utils/paginate';
 
-class ViewArticles extends Component {
+let articles;
+class AllArticles extends Component {
+  state = {
+    pageSize: 9,
+    currentPage: 1
+  }
+  componentDidMount() {
+    this.props.getArticles();
+  }
+
+  handleClick = slug => {
+    this.props.history.push(`/articles/${slug}`);
+  };
+
+  handlePageChange = page => {
+    this.setState({ currentPage: page });
+  };
   render() {
-    return (
-      <div className="container" id="component-ViewArticles">
-        Articles here
-      </div>
-    );
+    const { fetched } = this.props.state.articles;
+    const { pageSize, currentPage } = this.state;
+    let paginateArticles;
+    switch (fetched) {
+      case "":
+        return (
+          <div>
+            <NavBar />
+            <div className="masonry-container">
+              <div className="masonry">
+                <h2>loading...</h2>
+              </div>
+            </div>
+          </div>
+        );
+      case "pending":
+        return (
+          <div id="component-ViewArticles">
+            <NavBar />
+            <div className="masonry-container">
+              <div className="masonry">
+                <h2>loading...</h2>
+              </div>
+            </div>
+          </div>
+        );
+      case "done":
+        articles = this.props.state.articles.articles.data.articles;
+        paginateArticles = paginate(articles, currentPage, pageSize);
+        if (articles.length === 0) {
+          return (
+            <div>
+              <NavBar />
+              <div className="masonry-container">
+                <div className="masonry">No articles Available</div>
+              </div>
+            </div>
+          );
+        }
+        return (
+          <div>
+            <NavBar />
+            <div className="masonry-container">
+              <div className="masonry">
+                {paginateArticles.map(article => {
+                  return (
+                    <ArticleCard
+                      key={article.slug}
+                      article={article}
+                      onClick={this.handleClick}
+                    />
+                  );
+                })}
+              </div>
+            </div>
+            <Pagination
+              itemsCount={articles.length}
+              pageSize={pageSize}
+              currentPage={currentPage}
+              onPageChange={this.handlePageChange}
+            />
+          </div>
+        );
+      default:
```

Comment:
Why do you put `loading ` as default? I would suggest using `articles not available` or something.

Reply:
@AJAkimana actually this feature is for only paginating articles. the feature for viewing articles is for @jabichris and I will rebase from what he does. Thank you